File: 0042-trapping-rain-water/0042-trapping-rain-water.java
class Solution {
    public int trap(int[] height) {
        int low = 0;
        int high = height.length - 1;
        int lmax = height[low];
        int rmax = height[high];
        int res = 0;
        while(low < high){
            if(lmax < rmax){
                low++;
                lmax = Math.max(lmax, height[low]);
                res = res + lmax - height[low];
            }else{
                high--;
                rmax = Math.max(rmax, height[high]);
                res = res + rmax - height[high];
            }
        }
        return res;
    }
}

File: 0042-trapping-rain-water/0042-trapping-rain-water.py
class Solution:
    def trap(self, height: List[int]) -> int:
        low = 0
        high = len(height) - 1
        lmax = height[low]
        rmax = height[high]
        res = 0
        while low < high:
            if lmax < rmax:
                low += 1
                lmax = max(lmax, height[low])
                res += (lmax - height[low])
            else:
                high -= 1
                rmax = max(rmax, height[high])
                res += (rmax - height[high])
        return res